nav2_costmap_2d/plugins/costmap_filters/binary_filter.cpp

```diff
@@ -259,6 +295,61 @@ void BinaryFilter::changeState(const bool state)
     std::make_unique<std_msgs::msg::Bool>();
   msg->data = state;
   binary_state_pub_->publish(std::move(msg));
+  changeParameters(state);
+}
+
+void BinaryFilter::changeParameters(const bool state)
+{
+  for (size_t param_index = 0; param_index < binary_parameters_info_.size(); ++param_index) {
+    if (!change_parameters_clients_.at(param_index)->wait_for_service(
+        std::chrono::milliseconds(change_parameter_timeout_)))
+    {
+      RCLCPP_WARN(
+        logger_, "BinaryFilter:  service %s not available. Skipping ...",
+        change_parameters_clients_.at(param_index)->get_service_name());
+      continue;
+    } else {
+      RCLCPP_INFO(
+        logger_, "BinaryFilter:  service %s available.",
+        change_parameters_clients_.at(param_index)->get_service_name());
+    }
+
+    // Create a rcl_interfaces::msg::SetParameters client for changing parameters
+    auto request = std::make_shared<rcl_interfaces::srv::SetParameters::Request>();
+
+    // Set parameters for BinaryFilter
+    rcl_interfaces::msg::Parameter bool_param;
+    bool_param.name = binary_parameters_info_.at(param_index).param_name;
+    bool_param.value.type = rcl_interfaces::msg::ParameterType::PARAMETER_BOOL;
+    if (binary_parameters_info_.at(param_index).default_state != default_state_) {
+      bool_param.value.bool_value = !state;
+    } else {
+      bool_param.value.bool_value = state;
+    }
+    request->parameters.push_back(bool_param);
+
+    using ServiceResponseFuture =
+      rclcpp::Client<rcl_interfaces::srv::SetParameters>::SharedFuture;
+    auto response_received_callback = [this, bool_param](ServiceResponseFuture future) {
+        auto result = future.get();
+        if (!result->results.at(0).successful) {
+          RCLCPP_ERROR(
+            logger_, "BinaryFilter: Failed to change parameter %s",
+            bool_param.name.c_str());
+        } else {
+          RCLCPP_DEBUG(
+            logger_, "BinaryFilter: Successfully changed parameter to %s",
+            bool_param.value.bool_value ? "true" : "false");
+        }
+      };
+
+    RCLCPP_DEBUG(
+      logger_, "BinaryFilter: Sending request to set parameter  %s to %s",
+      binary_parameters_info_.at(param_index).param_name.c_str(),
+      bool_param.value.bool_value ? "true" : "false");
+    auto future_result = change_parameters_clients_.at(param_index)->async_send_request(
+      request, response_received_callback);
```

Comment:
Remove callback. Spin until future complete. Then check the value and throw something more substantial than a log if it failed

Reply:
I guess it depends on what you think about [this](https://github.com/ros-planning/navigation2/pull/3796#discussion_r1369208988)